Objects/enumobject.c

```diff
@@ -108,14 +104,50 @@ enumerate_vectorcall(PyObject *type, PyObject *const *args,
                     "'%S' is an invalid keyword argument for enumerate()", kw);
                 return NULL;
             }
+        } else if (nkwargs == 2) {
+            PyObject *kw0 = PyTuple_GET_ITEM(kwnames, 0);
+            PyObject *kw1 = PyTuple_GET_ITEM(kwnames, 1);
+            if (_PyUnicode_EqualToASCIIString(kw0, "start")) {
+                if (!_PyUnicode_EqualToASCIIString(kw1, "iterable")) {
+                    PyErr_Format(PyExc_TypeError,
+                        "'%S' is an invalid keyword argument for enumerate()", kw1);
+                    return NULL;
+                }
+                return enum_new_impl(tp, args[1], args[0]);
+            }
+            if (!_PyUnicode_EqualToASCIIString(kw0, "iterable")) {
```

Comment:
@JelleZijlstra 
Would you like to extract the function into `if (!checkKw(kwnames, 0, "iterable"))` other things too :)

Reply:
Good idea, that does make the code nicer